ExampleUsage/EnumExample.cs

```diff
@@ -55,20 +59,98 @@ public void ExampleOfUsingApply()
 				context.Database.ExecuteSqlCommand(migrationSql);
 			}
 		}
-	}
+
+        [Test]        
+        public void CheckIfDescriptionColumnsExists()
```

Comment:
You're right the test suite is a little inelegant at the moment and could do with tidying up and splitting out. I haven't got as far as thinking about the set of variations it needs to cover, it exists in its current form purely to enable me to test-drive the development to avoid regressions.